### DISCUSSION TITLE
1panel在商店点击安装Java运行环境就跳转到，运行环境页面，而不是应用安装页面，无法安装Java、Node、Go，这是为什么

### DISCUSSION THREAD

**Discussion by swordman20** [1Panel-dev/1Panel#6391] [Q&A]
### 联系方式

xiaweifeng@live.cn

### 1Panel 版本

v.1.10.14

### 问题描述

1panle在商店点击安装Java运行环境就跳转到，运行环境页面，而不是应用安装页面，无法安装Java、Node、Go，这是为什么。浏览Issues，有人反应这个问题，有人回复v.1.10.13已经解决了。可是我升级了，也还是这样的问题，不知道是不是bug

### 重现步骤

我本想在，网站->运行环境->创建Java、Node.js运行环境，创建完成启动总是失败，并且无日志描述。然后看论坛反应，应该需要从应用商店安装对应环境的Java和Node.js版本，才能创建运行环境。
我在1panle在商店点击安装Java运行环境就跳转到，运行环境页面，导致无法安装Java、Node.js。

### 期待的正确结果

1panle在商店点击安装Java，可以选择安装Java版本。并成功安装。

### 附加信息

![WX20240906-090905@2x](https://github.com/user-attachments/assets/0e1d9493-17fa-4a8b-b2fe-f2c169f8a678)
![WX20240906-091032@2x](https://github.com/user-attachments/assets/52e46cd3-7a5c-451c-a542-6b42fa5173e8)

**Comment by zhengkunwang223** [comment]
逻辑就是这样 可以先看一下我们的文档 https://1panel.cn/docs/user_manual/websites/node/

**Comment by swordman20** [comment]
> 逻辑就是这样可以先看一下我们的文档https://1panel.cn/docs/user_manual/websites/node/

我看了你们的文档，但是，首先是不是要在应用商店安装了对于的Java版本，才能运行环境里面的Java。问题一，我在应用商店里面无法安装Java。问题二，即使我无法安装Java、Node，我直接创建运行环境，确定后，启动直接报错。服务器内部错误，启动失败！

![WX20240906-095245@2x](https://github.com/user-attachments/assets/6847580e-4604-45e2-8f04-566036f77a01)

**Comment by zhengkunwang223** [comment]
> > 逻辑就是这样可以先看一下我们的文档https://1panel.cn/docs/user_manual/websites/node/
> 
> 我看了你们的文档，但是，首先是不是要在应用商店安装了对于的Java版本，才能运行环境里面的Java。问题一，我在应用商店里面无法安装Java。问题二，即使我无法安装Java、Node，我直接创建运行环境，确定后，启动直接报错。服务器内部错误，启动失败！
> 
> ![WX20240906-095245@2x](https://private-user-images.githubusercontent.com/7270004/365003436-6847580e-4604-45e2-8f04-566036f77a01.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU1ODgzOTMsIm5iZiI6MTcyNTU4ODA5MywicGF0aCI6Ii83MjcwMDA0LzM2NTAwMzQzNi02ODQ3NTgwZS00NjA0LTQ1ZTItOGYwNC01NjYwMzZmNzdhMDEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkwNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MDZUMDIwMTMzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OGMwZDcyNTBlYWM4YmI1OTEwYmYwMGVhNzdjMWZmY2EyODBiNTIyZmVhMmQ3ZDJkNTA0NWM0ODIwZTQxY2QyZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.zU4uY8Rd_Ju_gG6WM6PYCV0LrT0rzx2Ck0XzJoTNvBY)

1. Java Go Node PHP 等运行环境 都是要在运行环境页面安装的
2. 创建运行环境失败 需要看日志和错误信息 鼠标移动到失败状态上 可以看报错信息 点击日志按钮 也可以看运行中的报错信息

**Comment by swordman20** [comment]
> > > 逻辑就是这样可以先看一下我们的文档https://1panel.cn/docs/user_manual/websites/node/
> > 
> > 
> > 我看了你们的文档，但是，首先是不是要在应用商店安装了Java版本，才能在Java里面运行环境。问题一，我在应用商店里面安装无法Java。问题二，即使我无法安装Java 、Node，我直接创建运行环境，确定后，启动直接报错。服务器内部错误，启动失败！
> > ![WX20240906-095245@2x](https://private-user-images.githubusercontent.com/7270004/365003436-6847580e-4604-45e2-8f04-566036f77a01.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU1ODgzOTMsIm5iZiI6MTcyNTU4ODA5MywicGF0aCI6Ii83MjcwMDA0LzM2NTAwMzQzNi02ODQ3NTgwZS00NjA0LTQ1ZTItOGYwNC01NjYwMzZmNzdhMDEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkwNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MDZUMDIwMTMzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OGMwZDcyNTBlYWM4YmI1OTEwYmYwMGVhNzdjMWZmY2EyODBiNTIyZmVhMmQ3ZDJkNTA0NWM0ODIwZTQxY2QyZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.zU4uY8Rd_Ju_gG6WM6PYCV0LrT0rzx2Ck0XzJoTNvBY)
> 
> 1. Java Go Node PHP 等运行环境都是要在运行环境页面安装的
> 2. 创建运行环境失败 需要看日志和错误信息 鼠标移动到失败状态上 可以看报错信息 点击日志按钮 也可以看运行中的报错信息

感谢您的回复。
[我再次概述一下我的问题。相同类型问题[#5900](https://github.com/1Panel-dev/1Panel/issues/5900)。
**1.Java Go Node PHP 等运行环境都是要在运行环境页面安装的。**
答：我无法安装，它会直接跳转到，网站->运行环境的页面。

**2.创建运行环境失败 需要看日志和错误信息 鼠标移动到失败状态上 可以看报错信息 点击日志按钮 也可以看运行中的报错信息**
答：创建运行环境成功，但是启动失败，无任何错误日志。

![WX20240906-100945@2x](https://github.com/user-attachments/assets/913c1dfc-e99a-438c-8081-128986329d81)
![WX20240906-101029@2x](https://github.com/user-attachments/assets/d78e90d1-63f3-4073-89a6-5717fa1010e7)

您能理解了吗？我看过文档，也看过#5900反应的问题。我的问题和他的一样，我不知道为什么关闭了那个issues。我的1panle版本已经是最新的了。

**Comment by zhengkunwang223** [comment]
> > > > 逻辑就是这样可以先看一下我们的文档https://1panel.cn/docs/user_manual/websites/node/
> > > 
> > > 
> > > 我看了你们的文档，但是，首先是不是要在应用商店安装了Java版本，才能在Java里面运行环境。问题一，我在应用商店里面安装无法Java。问题二，即使我无法安装Java 、Node，我直接创建运行环境，确定后，启动直接报错。服务器内部错误，启动失败！
> > > ![WX20240906-095245@2x](https://private-user-images.githubusercontent.com/7270004/365003436-6847580e-4604-45e2-8f04-566036f77a01.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU1ODgzOTMsIm5iZiI6MTcyNTU4ODA5MywicGF0aCI6Ii83MjcwMDA0LzM2NTAwMzQzNi02ODQ3NTgwZS00NjA0LTQ1ZTItOGYwNC01NjYwMzZmNzdhMDEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkwNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MDZUMDIwMTMzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OGMwZDcyNTBlYWM4YmI1OTEwYmYwMGVhNzdjMWZmY2EyODBiNTIyZmVhMmQ3ZDJkNTA0NWM0ODIwZTQxY2QyZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.zU4uY8Rd_Ju_gG6WM6PYCV0LrT0rzx2Ck0XzJoTNvBY)
> > 
> > 
> > 
> > 1. Java Go Node PHP 等运行环境都是要在运行环境页面安装的
> > 2. 创建运行环境失败 需要看日志和错误信息 鼠标移动到失败状态上 可以看报错信息 点击日志按钮 也可以看运行中的报错信息
> 
> 感谢您的回复。 [我再次概述一下我的问题。相同类型问题[#5900](https://github.com/1Panel-dev/1Panel/issues/5900)。 **1.Java Go Node PHP 等运行环境都是要在运行环境页面安装的。** 答：我无法安装，它会直接跳转到，网站->运行环境的页面。
> 
> **2.创建运行环境失败 需要看日志和错误信息 鼠标移动到失败状态上 可以看报错信息 点击日志按钮 也可以看运行中的报错信息** 答：创建运行环境成功，但是启动失败，无任何错误日志。
> 
> ![WX20240906-100945@2x](https://private-user-images.githubusercontent.com/7270004/365008149-913c1dfc-e99a-438c-8081-128986329d81.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU1ODkxODgsIm5iZiI6MTcyNTU4ODg4OCwicGF0aCI6Ii83MjcwMDA0LzM2NTAwODE0OS05MTNjMWRmYy1lOTlhLTQzOGMtODA4MS0xMjg5ODYzMjlkODEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkwNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MDZUMDIxNDQ4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDU1NjU0YmRjZjkxMWEwMTA3MGU4MDdiZDczMDI2YzdmMTRhZjAxNGRiM2QxMTJjZTcwZWQyZWY2YTA3YjhiOSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.6eng5vVnj2XO-5trWyXxOclqg9KNGbqVydNs2tElIdM) ![WX20240906-101029@2x](https://private-user-images.githubusercontent.com/7270004/365008186-d78e90d1-63f3-4073-89a6-5717fa1010e7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU1ODkxODgsIm5iZiI6MTcyNTU4ODg4OCwicGF0aCI6Ii83MjcwMDA0LzM2NTAwODE4Ni1kNzhlOTBkMS02M2YzLTQwNzMtODlhNi01NzE3ZmExMDEwZTcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkwNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MDZUMDIxNDQ4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZGY2YjU2ZmU0OGQyY2I3YTY1Nzc4ZDY5MjQ1YWVlN2FkOTc2ZGY3NTJjZjI1YmNkMGZkNDY3YjNhNTQ4NTQxYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.Hl1bolcvxSMSyJSAc86ogH4sszIdfn4DGyEWdfkTOM0)
> 
> 您能理解了吗？我看过文档，也看过#5900反应的问题。我的问题和他的一样，我不知道为什么关闭了那个issues。我的1panle版本已经是最新的了。

1.  逻辑就是这么设计的 必须从 运行环境菜单安装 Java Node Go PHP 等运行环境
2. 没有日志和错误不太常见 启动的时候选 Node 项目的目录了吗

**Comment by swordman20** [comment]
> > > > > 逻辑就是这样可以先看一下我们的文档https://1panel.cn/docs/user_manual/websites/node/
> > > > 
> > > > 
> > > > 我看了你们的文档，但是，首先是不是要在应用商店里面安装了Java版本，才能在Java里面运行环境。问题一，我在应用商店里面安装无法Java。问题二，即使我无法安装Java、Node ，我直接创建运行环境，确定后，启动直接报错。服务器内部错误，启动失败！
> > > > ![WX20240906-095245@2x](https://private-user-images.githubusercontent.com/7270004/365003436-6847580e-4604-45e2-8f04-566036f77a01.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU1ODgzOTMsIm5iZiI6MTcyNTU4ODA5MywicGF0aCI6Ii83MjcwMDA0LzM2NTAwMzQzNi02ODQ3NTgwZS00NjA0LTQ1ZTItOGYwNC01NjYwMzZmNzdhMDEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkwNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MDZUMDIwMTMzWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OGMwZDcyNTBlYWM4YmI1OTEwYmYwMGVhNzdjMWZmY2EyODBiNTIyZmVhMmQ3ZDJkNTA0NWM0ODIwZTQxY2QyZSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.zU4uY8Rd_Ju_gG6WM6PYCV0LrT0rzx2Ck0XzJoTNvBY)
> > > 
> > > 
> > > 
> > > 1. Java Go Node PHP 等运行环境都是要在运行环境页面安装的
> > > 2. 创建运行环境失败 需要看日志和错误信息 鼠标移动到失败状态上 可以看报错信息 点击日志按钮 也可以看运行中的报错信息
> > 
> > 
> > 感谢您的回复。 [我再次概述了一下我的问题。相同类型问题[#5900](https://github.com/1Panel-dev/1Panel/issues/5900)。**1.Java Go Node PHP 等运行环境都是要在运行环境页面安装的。**答：我无法安装，它会直接跳转到，网站->运行环境的页面。
> > **2.运行环境失败 需要看日志和错误信息 鼠标移动到失败状态上 可以看报错信息 点击日志按钮 也可以看创建运行中的报错信息**答：创建运行环境成功，但是启动失败，无任何错误日志。
> > ![WX20240906-100945@2x](https://private-user-images.githubusercontent.com/7270004/365008149-913c1dfc-e99a-438c-8081-128986329d81.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU1ODkxODgsIm5iZiI6MTcyNTU4ODg4OCwicGF0aCI6Ii83MjcwMDA0LzM2NTAwODE0OS05MTNjMWRmYy1lOTlhLTQzOGMtODA4MS0xMjg5ODYzMjlkODEucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkwNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MDZUMDIxNDQ4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDU1NjU0YmRjZjkxMWEwMTA3MGU4MDdiZDczMDI2YzdmMTRhZjAxNGRiM2QxMTJjZTcwZWQyZWY2YTA3YjhiOSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.6eng5vVnj2XO-5trWyXxOclqg9KNGbqVydNs2tElIdM) ![WX20240906-101029@2x](https://private-user-images.githubusercontent.com/7270004/365008186-d78e90d1-63f3-4073-89a6-5717fa1010e7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU1ODkxODgsIm5iZiI6MTcyNTU4ODg4OCwicGF0aCI6Ii83MjcwMDA0LzM2NTAwODE4Ni1kNzhlOTBkMS02M2YzLTQwNzMtODlhNi01NzE3ZmExMDEwZTcucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkwNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MDZUMDIxNDQ4WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZGY2YjU2ZmU0OGQyY2I3YTY1Nzc4ZDY5MjQ1YWVlN2FkOTc2ZGY3NTJjZjI1YmNkMGZkNDY3YjNhNTQ4NTQxYiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.Hl1bolcvxSMSyJSAc86ogH4sszIdfn4DGyEWdfkTOM0)
> > 您能理解吗？我看过文档，也看过#5900反应的问题。我的问题和他的一样，我不知道为什么关闭了那个问题。我的1panle版本已经是最新的了。
> 
> 1. 逻辑就是这么设计的 必须从运行环境菜单安装 Java Node Go PHP 等运行环境
> 2. 没有日志和错误不太常见 启动的时候选择 Node 项目的目录了吗

我作为一名程序员，我来回答一下你的回复，并重复提出我的问题。
**1. 逻辑就是这么设计的 必须从运行环境菜单安装 Java Node Go PHP 等运行环境**
答：你应该说，“1panle，运行环境，依赖于应用商店安装的各版本，Java、Node.js、Go。你首先需要从应用商店安装对应的程序。才能在网站->运行环境里面，创建已经下载的程序，运行。”
**我的问题是，我压根无法在应用商店安装运行环境的程序。Java、Go、Node.js都无法安装**

**2. 没有日志和错误不太常见 启动的时候选择 Node 项目的目录了吗**
答：这个可以理解成，你连应用程序都没安装，就创建运行指令，那肯定失败。至于没有日志，那肯定是，1panle设计，只输出Java、Node应用程序的日志。你连应用程序都没有，那里来的错误日志。
当然这个1panle可以优化一下，如果判断没有，下载对应的Java、Node等软件，就无法创建运行环境指令。

您好，请不要复制粘贴回复，“逻辑就是这么设计的，你看一下文档 ”之类的话语，我表示，看过文档。如果您这边不知道问题为什么这样。请让其他人来回复一下。

**Comment by see-more** [comment]
你的应用程序是什么🤔

**Reply by swordman20** [reply]
是一个vue 的前端网页，使用了node.js编译。但是，启动总是报，服务内部错误: 启动失败:。并且无错误日志输出。
<img width="840" alt="Snipaste_2024-09-11_23-31-52" src="https://github.com/user-attachments/assets/bebb0ba2-3d79-41e6-860b-6feec946dd37">

<img width="1457" alt="Snipaste_2024-09-11_23-31-24" src="https://github.com/user-attachments/assets/4fd26152-0b21-4d4e-8876-899c1ff0d356">

**Reply by see-more** [reply]
`ng serve`?这个是angular的命令行吧 你确定这个项目你在本地可以正常运行么

**Reply by swordman20** [reply]
这个项目，在我本地可以运行。我除了Node.js报错，我的Java运行环境也报错。服务内部错误: 启动失败:。可以说，运行环境，我就使用不了，不知道为什么。错误日志也同样没有。
<img width="1503" alt="Snipaste_2024-09-11_23-53-58" src="https://github.com/user-attachments/assets/ace31329-1d40-4e4a-a3a0-9fd1bd3e6cd0">

<img width="803" alt="Snipaste_2024-09-11_23-54-19" src="https://github.com/user-attachments/assets/80bfec23-f9a9-4614-9eb0-d31a1dedaab0">

**Reply by see-more** [reply]
那方便给一个在你的服务器上无法运行的项目 我复现一下

**Reply by swordman20** [reply]
好的，我认为可能不是项目的问题，可能是1panel，安装环境的问题。因为我的Java运行环境也无法启动。
[前端本地导航项目nav](https://github.com/xjh22222228/nav)
[这是我要部署的java -jar项目，本地打包好，放服务器上MCMS](https://github.com/ming-soft/MCMS)。
以上两个项目，我在虚拟机的**CentOS Stream release 9**，上安装了1panel面板，并使用运行环境，可以运行。
但是，在我的NAT的**CentOS Stream release 8**上，1panel，运行环境，就会报错。

**Reply by swordman20** [reply]
我不知道，是不是我在安装1panel之后，做了哪些设置，导致了，运行环境无法使用。还是因为什么。我在虚拟机上，直接在线安装1panel，未做相关设置，并直接使用运行环境。可以启动相关node、java程序，即使失败，也有log日志输出。

**Reply by liujan** [reply]
你好，请问有找到原因吗？我的也是同样的错误，也没有日志，不知道为什么报错，也不知道怎么解

**Reply by swordman20** [reply]
换系统，使用centos7，不要用stream

**Reply by LH-web** [reply]
你好，我是安装飞牛nas中安装1panel，去部署nodejs环境，也出现了启动，模块无法安装，无日志的情况

**Reply by SilentQing** [reply]
> 你好，我是安装飞牛nas中安装1panel，去部署nodejs环境，也出现了启动，模块无法安装，无日志的情况

通过在终端使用，独立安装的docker compose，docker-compose来执行的试试，不要用apt安装方式（插件安装方式）安装，用二进制文件（独立安装）试试

**Comment by zhengkunwang223** [comment]
1.飞牛中安装的 1panel 需要先手动安装 docker-compose 
2. 建议使用一键安装脚本在飞牛中安装 1panel 参考  https://1panel.cn/docs/installation/online_installation/
